Entitas/Entitas/Interfaces/IEntity.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Entitas {
+
+    public delegate void EntityChanged(Entity entity, int index, IComponent component);
+    public delegate void ComponentReplaced(Entity entity, int index, IComponent previousComponent, IComponent newComponent);
+    public delegate void EntityReleased(Entity entity);
```

Comment:
Before these types were declared inside the `Entity` class. Since interfaces can't declare types I moved them here. Should I put them in their own namespace or leave them in the global Entitas namespace?